pkg/tagger/collectors/docker_extract_test.go

```diff
@@ -0,0 +1,130 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build docker
+
+package collectors
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDockerRecordsFromInspect(t *testing.T) {
+	testCases := []struct {
+		co                   *types.ContainerJSON
+		toRecordEnvAsTags    map[string]string
+		toRecordLabelsAsTags map[string]string
+		expectedLow          []string
+		expectedHigh         []string
+	}{
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelValue"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{},
+			toRecordLabelsAsTags: map[string]string{},
+			expectedLow:          []string{},
+			expectedHigh:         []string{},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelKey"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "becomeK"},
+			toRecordLabelsAsTags: map[string]string{"labelKey": "labelValue"},
+			expectedLow:          []string{"becomeK:v"},
+			expectedHigh:         []string{},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelKey"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "+becomeK"},
+			toRecordLabelsAsTags: map[string]string{"labelKey": "labelValue"},
```

Comment:
We lowercase all keys from the conf in import, so you should never get mixed case `labelKey` in the conf dict. This is un-intuitive, as I'd expect to match if I put mixed case in the yaml conf